src/main/lib/icon/storage/IconFileStorage.js

```diff
@@ -0,0 +1,103 @@
+import fsLib from 'fs';
+import mkdirpLib from 'mkdirp';
+import pify from 'pify';
+import path from 'path';
+import sanitize from 'sanitize-filename';
+import { StorageInterface } from '@buttercup/iconographer';
+
+const fs = pify(fsLib);
+const mkdirp = pify(mkdirpLib);
+
+export default class IconFileStorage extends StorageInterface {
+  constructor(path) {
+    super();
+    this.path = path;
+  }
+
+  _buildKeyPath(iconKey) {
+    // TODO No extension?
+    return path.join(this.path, sanitize(iconKey)) + '.ico';
```

Comment:
TBD

Reply:
I would recommend an extension of `.dat`, for instance, to prevent any arbitrary assignment of type (eg ico/png/jpg etc.).

Reply:
^

Reply:
Maybe even `.icon.dat`, for instance..